--- xfce4-trigger-launcher-4.1.90/configure.ac.orig	2004-10-02 13:58:32.000000000 +0200
+++ xfce4-trigger-launcher-4.1.90/configure.ac	2004-10-08 06:36:46.554179544 +0200
@@ -32,8 +32,8 @@
 AC_CHECK_FUNCS([sigaction])
 
 dnl Check for i18n support
-BM_I18N([xfce4-trigger-launcher], [ar az be bg ca cs de es et eu fa_IR fr gu he
-				   hi hu it ja ko lt ms nl pl pt_BR pt_PT
+BM_I18N([xfce4-trigger-launcher], [ar az be bg ca cs de es et eu fa fr gu he
+				   hi hu it ja ko lt ms nl pl pt_BR pt
 				   ru sv sk ta tr uk vi zh_CN zh_TW])
 
 dnl configure the panel plugin
