--- xfce4-trigger-launcher-4.1.91/configure.ac.orig	2004-10-22 21:36:34.000000000 +0200
+++ xfce4-trigger-launcher-4.1.91/configure.ac	2004-11-13 20:57:30.935337432 +0100
@@ -34,7 +34,7 @@
 dnl Check for i18n support
 BM_I18N([xfce4-trigger-launcher], [ar az be bg ca cs de es es_MX et eu fa fi fr
 				   gu he hi hu it ja ko lt ms nl pa pl pt_BR
-				   pt_PT ru sv sk ta tr uk vi zh_CN zh_TW])
+				   pt ru sv sk ta tr uk vi zh_CN zh_TW])
 
 dnl configure the panel plugin
 XFCE_PANEL_PLUGIN([XFCE4_PANEL], [4.0.0])
