diff -urN xfce4-trigger-launcher-4.2.4.1-orig/config.h.in xfce4-trigger-launcher-4.2.4.1/config.h.in
--- xfce4-trigger-launcher-4.2.4.1-orig/config.h.in	2007-01-28 11:32:03.000000000 +0100
+++ xfce4-trigger-launcher-4.2.4.1/config.h.in	2011-03-01 13:44:42.000000000 +0100
@@ -63,6 +63,10 @@
 /* Define to 1 if you have the <unistd.h> header file. */
 #undef HAVE_UNISTD_H
 
+/* Define to the sub-directory in which libtool stores uninstalled libraries.
+   */
+#undef LT_OBJDIR
+
 /* Name of package */
 #undef PACKAGE
 
@@ -78,21 +82,39 @@
 /* Define to the one symbol short name of this package. */
 #undef PACKAGE_TARNAME
 
+/* Define to the home page for this package. */
+#undef PACKAGE_URL
+
 /* Define to the version of this package. */
 #undef PACKAGE_VERSION
 
 /* Define to 1 if you have the ANSI C header files. */
 #undef STDC_HEADERS
 
-/* Version number of package */
-#undef VERSION
-
-/* Define to 1 if on AIX 3.
-   System headers sometimes define this.
-   We just want to avoid a redefinition error message.  */
+/* Enable extensions on AIX 3, Interix.  */
 #ifndef _ALL_SOURCE
 # undef _ALL_SOURCE
 #endif
+/* Enable GNU extensions on systems that have them.  */
+#ifndef _GNU_SOURCE
+# undef _GNU_SOURCE
+#endif
+/* Enable threading extensions on Solaris.  */
+#ifndef _POSIX_PTHREAD_SEMANTICS
+# undef _POSIX_PTHREAD_SEMANTICS
+#endif
+/* Enable extensions on HP NonStop.  */
+#ifndef _TANDEM_SOURCE
+# undef _TANDEM_SOURCE
+#endif
+/* Enable general extensions on Solaris.  */
+#ifndef __EXTENSIONS__
+# undef __EXTENSIONS__
+#endif
+
+
+/* Version number of package */
+#undef VERSION
 
 /* Define to 1 if on MINIX. */
 #undef _MINIX
diff -urN xfce4-trigger-launcher-4.2.4.1-orig/configure.in xfce4-trigger-launcher-4.2.4.1/configure.in
--- xfce4-trigger-launcher-4.2.4.1-orig/configure.in	2011-03-01 11:20:45.745067638 +0100
+++ xfce4-trigger-launcher-4.2.4.1/configure.in	2011-03-01 13:44:38.973063514 +0100
@@ -35,7 +35,7 @@
 
 dnl configure the panel plugin
 XDT_CHECK_PACKAGE([XFCE4_PANEL], [libxfce4panel-1.0], [4.4.0])
-XDT_CHECK_PACKAGE([XFCE4_GUI], [libxfcegui4-1.0], [4.4.0])
+XDT_CHECK_PACKAGE([LIBXFCE4UI], [libxfce4ui-1], [4.8.0])
 
 dnl Check for debugging support
 BM_DEBUG_SUPPORT
diff -urN xfce4-trigger-launcher-4.2.4.1-orig/configure.in.in xfce4-trigger-launcher-4.2.4.1/configure.in.in
--- xfce4-trigger-launcher-4.2.4.1-orig/configure.in.in	2007-01-28 11:29:03.000000000 +0100
+++ xfce4-trigger-launcher-4.2.4.1/configure.in.in	2011-03-01 13:44:38.974063514 +0100
@@ -35,7 +35,7 @@
 
 dnl configure the panel plugin
 XDT_CHECK_PACKAGE([XFCE4_PANEL], [libxfce4panel-1.0], [4.4.0])
-XDT_CHECK_PACKAGE([XFCE4_GUI], [libxfcegui4-1.0], [4.4.0])
+XDT_CHECK_PACKAGE([LIBXFCE4UI], [libxfce4ui-1], [4.8.0])
 
 dnl Check for debugging support
 BM_DEBUG_SUPPORT
diff -urN xfce4-trigger-launcher-4.2.4.1-orig/panel-plugin/Makefile.am xfce4-trigger-launcher-4.2.4.1/panel-plugin/Makefile.am
--- xfce4-trigger-launcher-4.2.4.1-orig/panel-plugin/Makefile.am	2007-01-28 11:29:02.000000000 +0100
+++ xfce4-trigger-launcher-4.2.4.1/panel-plugin/Makefile.am	2011-03-02 00:57:43.717634344 +0100
@@ -7,7 +7,7 @@
 	-export-dynamic							\
 	-module \
 	@XFCE4_PANEL_LIBS@                                              \
-	@XFCE4_GUI_LIBS@
+	@LIBXFCE4UI_LIBS@
 
 xfce4_trigger_launcher_plugin_SOURCES =                                         \
 	triggerlauncher.c                                               \
@@ -15,6 +15,8 @@
 	xfce-launcher-command-entry.h                                   \
 	xfce-trigger-launcher-options.c                                 \
 	xfce-trigger-launcher-options.h                                 \
+	xfce-iconbutton.c                                      \
+	xfce-iconbutton.h                                      \
 	xfce-file-chooser-button.c                                      \
 	xfce-file-chooser-button.h                                      \
 	xfce-icon-chooser-button.c                                      \
@@ -24,7 +26,7 @@
 	-DPACKAGE_LOCALE_DIR=\"$(localedir)\"				\
 	-DLOCALEDIR=\"$(localedir)\"                                    \
 	@XFCE4_PANEL_CFLAGS@                                            \
-	@XFCE4_GUI_CFLAGS@
+	@LIBXFCE4UI_CFLAGS@
 
 # .desktop file
 #
diff -urN xfce4-trigger-launcher-4.2.4.1-orig/panel-plugin/triggerlauncher.c xfce4-trigger-launcher-4.2.4.1/panel-plugin/triggerlauncher.c
--- xfce4-trigger-launcher-4.2.4.1-orig/panel-plugin/triggerlauncher.c	2007-01-28 11:29:02.000000000 +0100
+++ xfce4-trigger-launcher-4.2.4.1/panel-plugin/triggerlauncher.c	2011-03-02 01:02:32.607308341 +0100
@@ -11,10 +11,11 @@
 #include <signal.h>
 
 #include <gtk/gtk.h>
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 
-#include <libxfce4panel/xfce-panel-plugin.h>
+#include <libxfce4panel/libxfce4panel.h>
 #include "xfce-trigger-launcher-options.h"
+#include "xfce-iconbutton.h"
               
 #define PLUGIN_NAME "triggerlauncher"
 #define TRIGGER_LAUNCHER_RC_GROUP "Trigger Launcher"
@@ -129,7 +130,7 @@
     free_pixbufs (data1);
     
     
-    data1->fallback_pixbuf = xfce_themed_icon_load ("xfce-unknown", data1->pixbuf_size);  
+    data1->fallback_pixbuf = gtk_icon_theme_load_icon ( gtk_icon_theme_get_default(), "xfce-unknown", data1->pixbuf_size, GTK_ICON_LOOKUP_GENERIC_FALLBACK, NULL);
     data1->enabled_pixbuf = gdk_pixbuf_new_from_file_at_size (data1->enabled_icon, 
                                                               data1->pixbuf_size, 
                                                               data1->pixbuf_size,
diff -urN xfce4-trigger-launcher-4.2.4.1-orig/panel-plugin/xfce-iconbutton.c xfce4-trigger-launcher-4.2.4.1/panel-plugin/xfce-iconbutton.c
--- xfce4-trigger-launcher-4.2.4.1-orig/panel-plugin/xfce-iconbutton.c	1970-01-01 01:00:00.000000000 +0100
+++ xfce4-trigger-launcher-4.2.4.1/panel-plugin/xfce-iconbutton.c	2011-03-02 00:59:54.730634343 +0100
@@ -0,0 +1,100 @@
+/* $Id$
+ * 
+ * Copyright 2002-2004 Jasper Huijsmans (jasper@xfce.org)
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public
+ * License along with this library; if not, write to the
+ * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ */
+
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#include <math.h>
+
+#include <gdk/gdk.h>
+#include <gtk/gtkbutton.h>
+
+#include "libxfce4panel/libxfce4panel.h"
+#include "xfce-iconbutton.h"
+
+#define DEBUGGING 0
+
+/* XXX: this is used by gtkbutton. hope they won't change it soon ... */
+#define CHILD_SPACING 1
+
+G_DEFINE_TYPE (XfceIconbutton, xfce_iconbutton, GTK_TYPE_BUTTON)
+
+/* workaround for http://bugzilla.gnome.org/show_bug.cgi?id=142417 */
+static void
+_style_set_cb(GtkWidget *w, GtkStyle *prev_style, gpointer user_data)
+{
+    GtkStyle *style;
+
+    if (XFCE_ICONBUTTON (w)->image)
+    {
+        style = gtk_rc_get_style_by_paths (gtk_settings_get_default (),
+            "GtkButton", "GtkButton", GTK_TYPE_BUTTON);
+        gtk_widget_set_style (XFCE_ICONBUTTON (w)->image, style);
+    }
+}
+
+static void
+xfce_iconbutton_class_init (XfceIconbuttonClass * class)
+{
+}
+
+static void
+xfce_iconbutton_init (XfceIconbutton * iconbutton)
+{
+    iconbutton->image = xfce_panel_image_new ();
+    gtk_widget_show (iconbutton->image);
+    gtk_container_add (GTK_CONTAINER (iconbutton), iconbutton->image);
+
+    iconbutton->icon_width = iconbutton->icon_height = -1;
+    iconbutton->pb = NULL;
+}
+
+GtkWidget *
+xfce_iconbutton_new (void)
+{
+    GtkWidget *iconbutton = GTK_WIDGET (g_object_new (xfce_iconbutton_get_type (), NULL));
+
+    g_signal_connect (G_OBJECT (iconbutton), "style-set",
+        G_CALLBACK (_style_set_cb), NULL);
+
+    return iconbutton;
+}
+
+GtkWidget *
+xfce_iconbutton_new_from_pixbuf (GdkPixbuf * pixbuf)
+{
+    GtkWidget *iconbutton;
+
+    iconbutton = xfce_iconbutton_new ();
+    xfce_iconbutton_set_pixbuf (XFCE_ICONBUTTON (iconbutton), pixbuf);
+
+    return iconbutton;
+}
+
+void
+xfce_iconbutton_set_pixbuf (XfceIconbutton * iconbutton, GdkPixbuf * pixbuf)
+{
+    g_return_if_fail (iconbutton != NULL);
+    g_return_if_fail (XFCE_IS_ICONBUTTON (iconbutton));
+
+    xfce_panel_image_set_from_pixbuf (XFCE_PANEL_IMAGE (iconbutton->image),
+	    			       pixbuf);
+}
diff -urN xfce4-trigger-launcher-4.2.4.1-orig/panel-plugin/xfce-iconbutton.h xfce4-trigger-launcher-4.2.4.1/panel-plugin/xfce-iconbutton.h
--- xfce4-trigger-launcher-4.2.4.1-orig/panel-plugin/xfce-iconbutton.h	1970-01-01 01:00:00.000000000 +0100
+++ xfce4-trigger-launcher-4.2.4.1/panel-plugin/xfce-iconbutton.h	2011-03-02 00:57:12.278634342 +0100
@@ -0,0 +1,65 @@
+/*  xfce4
+ *  Copyright (C) 2002 Olivier Fourdan (fourdan@xfce.org)
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public
+ * License along with this library; if not, write to the
+ * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ */
+
+#ifndef __XFCE_ICONBUTTON_H__
+#define __XFCE_ICONBUTTON_H__
+
+#include <gdk/gdk.h>
+#include <gtk/gtkbutton.h>
+#include <gtk/gtkenums.h>
+
+
+G_BEGIN_DECLS
+
+#define XFCE_TYPE_ICONBUTTON (xfce_iconbutton_get_type ())
+#define XFCE_ICONBUTTON(obj)          G_TYPE_CHECK_INSTANCE_CAST (obj, xfce_iconbutton_get_type (), XfceIconbutton)
+#define XFCE_ICONBUTTON_CLASS(klass)  G_TYPE_CHECK_CLASS_CAST (klass, xfce_iconbutton_get_type (), XfceIconbuttonClass)
+#define XFCE_IS_ICONBUTTON(obj)       G_TYPE_CHECK_INSTANCE_TYPE (obj, xfce_iconbutton_get_type ())
+
+    typedef struct _XfceIconbutton XfceIconbutton;
+    typedef struct _XfceIconbuttonClass XfceIconbuttonClass;
+
+    struct _XfceIconbutton
+    {
+	GtkButton button;
+
+	GtkWidget *image;
+	GdkPixbuf *pb;
+
+        gint icon_width;
+        gint icon_height;
+    };
+
+    struct _XfceIconbuttonClass
+    {
+	GtkButtonClass parent_class;
+    };
+
+
+    GtkType xfce_iconbutton_get_type (void);
+    GtkWidget *xfce_iconbutton_new (void);
+    GtkWidget *xfce_iconbutton_new_from_pixbuf (GdkPixbuf * pb);
+
+    void xfce_iconbutton_set_pixbuf (XfceIconbutton * icon_button,
+				     GdkPixbuf * pixbuf);
+
+G_END_DECLS
+
+#endif				/* __XFCE_ICONBUTTON_H__ */
+/* example-end */
diff -urN xfce4-trigger-launcher-4.2.4.1-orig/panel-plugin/xfce-launcher-command-entry.c xfce4-trigger-launcher-4.2.4.1/panel-plugin/xfce-launcher-command-entry.c
--- xfce4-trigger-launcher-4.2.4.1-orig/panel-plugin/xfce-launcher-command-entry.c	2007-01-28 11:29:02.000000000 +0100
+++ xfce4-trigger-launcher-4.2.4.1/panel-plugin/xfce-launcher-command-entry.c	2011-03-01 13:44:38.978063514 +0100
@@ -5,7 +5,7 @@
 #include <string.h>
 #endif /* HAVE_STRING_H */
 #include <libxfce4util/libxfce4util.h>
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 #include <assert.h>
 #include <ctype.h>
 #include "xfce-launcher-command-entry.h"
diff -urN xfce4-trigger-launcher-4.2.4.1-orig/panel-plugin/xfce-trigger-launcher-options.c xfce4-trigger-launcher-4.2.4.1/panel-plugin/xfce-trigger-launcher-options.c
--- xfce4-trigger-launcher-4.2.4.1-orig/panel-plugin/xfce-trigger-launcher-options.c	2007-01-28 11:29:02.000000000 +0100
+++ xfce4-trigger-launcher-4.2.4.1/panel-plugin/xfce-trigger-launcher-options.c	2011-03-02 01:00:46.668634343 +0100
@@ -3,7 +3,7 @@
 #endif
 
 #include <gtk/gtk.h>
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 #include <libxfce4util/libxfce4util.h>
 #include "xfce-trigger-launcher-options.h"
 #include "xfce-launcher-command-entry.h"
@@ -233,11 +233,11 @@
    
 
 
-  check_status_header = xfce_create_framebox_with_content (_("To Check Status"), GTK_WIDGET(check_status_box));
-  when_disabled_header = xfce_create_framebox_with_content (_("When Disabled"), GTK_WIDGET(when_disabled_box));
-  when_enabled_header = xfce_create_framebox_with_content (_("When Enabled"), GTK_WIDGET(when_enabled_box));
-  when_undefined_header = xfce_create_framebox_with_content (_("When Undefined"), GTK_WIDGET(when_undefined_box));
-  when_dodgy_header = xfce_create_framebox_with_content (_("When Dodgy"), GTK_WIDGET(when_dodgy_box));
+  check_status_header = xfce_gtk_frame_box_new_with_content (_("To Check Status"), GTK_WIDGET(check_status_box));
+  when_disabled_header = xfce_gtk_frame_box_new_with_content (_("When Disabled"), GTK_WIDGET(when_disabled_box));
+  when_enabled_header = xfce_gtk_frame_box_new_with_content (_("When Enabled"), GTK_WIDGET(when_enabled_box));
+  when_undefined_header = xfce_gtk_frame_box_new_with_content (_("When Undefined"), GTK_WIDGET(when_undefined_box));
+  when_dodgy_header = xfce_gtk_frame_box_new_with_content (_("When Dodgy"), GTK_WIDGET(when_dodgy_box));
   
   gtk_widget_show (GTK_WIDGET(check_status_header));
   gtk_widget_show_all (GTK_WIDGET(check_status_command_box));
