--- xfce4-trigger-launcher-4.2.3.orig/configure.ac	2005-11-05 18:10:41.000000000 +0100
+++ xfce4-trigger-launcher-4.2.3/configure.ac	2006-07-05 12:46:20.328055000 +0200
@@ -35,7 +35,7 @@
 BM_I18N([xfce4-trigger-launcher], [af ar az be bg bn_IN ca cs da de el 
 				   en_GB eo es es_MX et eu fa fi fr gu
 				   he hi hu it ja ko ku lt mr ms nl pa
-				   pl pt_BR pt_PT ro ru sk sl sv ta tr
+				   pl pt_BR pt ro ru sk sl sv ta tr
 				   uk vi zh_CN zh_TW])
 
 dnl configure the panel plugin
